packages/apps/graph/src/services/node-service.ts

```diff
@@ -0,0 +1,56 @@
+import type { ChainId, IClient } from '@kadena/client';
+import { createClient, Pact } from '@kadena/client';
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type ChainModuleAccountDetails = {
+  account: string;
+  balance: number;
+  guard: {
+    keys: string[];
+    pred: 'keys-all' | 'keys-any' | 'keys-two';
+  };
+};
+
+function getClient(chainId: string): IClient {
+  return createClient(
+    `http://${process.env.NETWORK_HOST}/chainweb/0.0/${process.env.NETWORK_ID}/chain/${chainId}/pact`,
+  );
+}
+
+export async function getAccountDetails(
+  module: string,
+  accountName: string,
+  chainId: string,
+): Promise<ChainModuleAccountDetails> {
+  const commandResult = await getClient(chainId).local(
+    Pact.builder
+      .execution(
+        // @ts-ignore
+        Pact.modules[module].details(accountName),
+      )
+      .setMeta({
+        chainId: chainId as ChainId,
+      })
+      .setNetworkId(process.env.NETWORK_ID as string)
+      .createTransaction(),
+    {
+      preflight: false,
+    },
+  );
```

Comment:
Maybe here we could use `dirtyRead`?

```suggestion
    const commandResult = await getClient(chainId).dirtyRead(
    Pact.builder
      .execution(
        // @ts-ignore
        Pact.modules[module].details(accountName),
      )
      .setMeta({
        chainId: chainId as ChainId,
      })
      .setNetworkId(process.env.NETWORK_ID as string)
      .createTransaction(),
  );
```

Reply:
Fixed 👍